src/test/java/underlay/RMIUnderlayTest.java

```diff
@@ -0,0 +1,117 @@
+package underlay;
+
+import blockchain.LightChainInterface;
+import blockchain.Transaction;
+import fixture.Fixtures;
+import org.junit.jupiter.api.Test;
+import signature.SignedBytes;
+import skipGraph.NodeInfo;
+import skipGraph.SkipNodeInterface;
+import underlay.rmi.RMIUnderlay;
+import underlay.requests.skipgraph.GetLeftNodeRequest;
+import underlay.requests.lightchain.GetPublicKeyRequest;
+import underlay.requests.skipgraph.GetRightNumIDRequest;
+import underlay.requests.lightchain.PoVRequest;
+import underlay.responses.NodeInfoResponse;
+import underlay.responses.IntegerResponse;
+import underlay.responses.PublicKeyResponse;
+import underlay.responses.SignatureResponse;
+
+import java.io.FileNotFoundException;
+import java.rmi.RemoteException;
+import java.security.PublicKey;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static underlay.responses.IntegerResponse.IntegerResponseOf;
+import static underlay.responses.NodeInfoResponse.NodeInfoResponseOf;
+import static underlay.responses.PublicKeyResponse.PublicKeyResponseOf;
+import static underlay.responses.SignatureResponse.SignatureResponseOf;
+
+/**
+ * RMIUnderlay Unit Tests to make sure messages are sent/received correctly. The tests are done
+ * through mocking.
+ */
+public class RMIUnderlayTest {
+  private Fixtures fixtures = new Fixtures();
+  /**
+   * A test for the GetRightNumIDRequest message for the RMIUnderlay
+   *
+   * @throws RemoteException
+   * @throws FileNotFoundException
+   */
+}
+
+//    @Test
```

Comment:
Two comments about this file: 
- We should not have any commented code merged with master. If we are not sure about a code snippet, we should completely remove it instead of commenting it out.
- Every new feature (e.g., `RMIUnderlay`) we add, should be supported with unit tests. Now, are we losing unit test support for `RMIUnderlay`?

Reply:
Implemented in https://github.com/yhassanzadeh13/lightchain-container/pull/15/commits/e97c50a210f563c29f346cb17288ab76631f533c